### PR TITLE
dutch exchange testing setup

### PR DESCRIPTION
**Pull request by mikec**
@emaG3m this separates unit tests from "scenario" tests. using the `test/scenarios` dir to keep scripts that run against the full set of smart contracts deployed via migrations. changed the name of `run_futarchy` to `addListing`, but it's more or less the same code. Lots of commented stuff in there right now since it's in progress .. just want to get our changes synced up.

### CODE REVIEW

File: contracts/Challenge/FutarchyChallengeFactory.sol [hunk 1]
```diff
@@ -96,16 +96,24 @@ contract FutarchyChallengeFactory is ChallengeFactoryInterface {
     );
   }
 
+  event Debug(address token1, address token2, uint currentAuctionIndex);
```
Comment (emaG3m) [inline contracts/Challenge/FutarchyChallengeFactory.sol]: do we want to get rid of this?

File: contracts/Challenge/FutarchyChallengeFactory.sol [hunk 2]
```diff
@@ -96,16 +96,24 @@ contract FutarchyChallengeFactory is ChallengeFactoryInterface {
...
     uint firstReferencedIndex = currentAuctionIndex - NUM_PRICE_POINTS;
 
     uint i = 0;
     uint num;
     uint den;
     uint avgPrice;
     while(i < NUM_PRICE_POINTS) {
-      (num, den) = dutchExchange.getPriceInPastAuction(token, comparatorToken, firstReferencedIndex + i);
+      // (num, den) = dutchExchange.getPriceInPastAuction(token, comparatorToken, firstReferencedIndex + i);
+      num = 10;
```
Comment (emaG3m) [inline contracts/Challenge/FutarchyChallengeFactory.sol]: does this work without hard-coded settings?

File: migrations/2_2_deploy_dx_mock.js [hunk 3]
```diff
@@ -19,8 +19,10 @@ const OWLAirdrop = artifacts.require('OWLAirdrop')
 let currentETHPrice = (1100 * (10 ** 18))
 
 module.exports = (deployer, network, accounts) => {
+  console.log('DXMOCK')
```
Comment (emaG3m) [inline migrations/2_2_deploy_dx_mock.js]: extra `console.logs` in this file?

File: scripts/scenarios.sh [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+#   - account params from a JSON file
+#   - `testrpc start` and `testrpc stop`, so these custom scripts don't
+#      have to be added to every project
+
+output=$(nc -z localhost 8545; echo $?)
+[ $output -eq "0" ] && ganache_running=true
+if [ ! $ganache_running ]; then
+  echo "Starting our own ganache instance"
+  ganache-cli \
+    -l 7000000 \
```
Comment (emaG3m) [inline scripts/scenarios.sh]: is this gas limit?

File: test/helpers/distributeToken.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+const amount = 100 * 10 ** 18
```
Comment (emaG3m) [inline test/helpers/distributeToken.js]: why should these be separate from `test/utils`?

**Review by ewilz — CHANGES_REQUESTED** [review]

File: scripts/scenarios.sh (hunk 4 above)
Comment (mikec) [reply scripts/scenarios.sh]: yep, I think this came from the setup we had before the proxy optimizations

**Comment by mikec** [comment]
@emaG3m, changes made. we might want to rethink putting all test helpers in a single file (utils.js), but for consistency I moved the helpers in there.

**Review by ewilz — APPROVED** [review]

**Pull request merged**